core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -788,6 +788,20 @@ abstract class RDD[T: ClassTag](
   }
 
   /**
+   * Applies a function to each partition of this RDD, while tracking the index
+   * of the original partition.
+   */
+  def foreachPartitionWithIndex(
```

Comment:
Yeah, I was originally doing map followed by empty foreach and thought it looked confusing.

I think it's really a question of what's easier to explain, it's just a syntax sugar issue not a correctness issue, so no problem either way.